### PR TITLE
ref(debuginfo): Switch to error with kind

### PR DESCRIPTION
**Pull request by jan-auer**
Follows the conventions set by `std::io::Error`.

### CODE REVIEW

**Review by Swatinem — APPROVED** [review]
Haven’t looked at the descriptions & docs in detail, high level looks very good, and getting rid of the details in `ObjectError` sounds good

File: symbolic-debuginfo/src/object.rs [hunk 1]
```diff
@@ -49,58 +50,75 @@ macro_rules! map_inner {
...
+}
 
-    /// An error in DWARF debugging information.
-    #[error("failed to process dwarf info")]
-    Dwarf(#[from] DwarfError),
+impl fmt::Debug for ObjectError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.repr {
+            ObjectErrorRepr::Transparent(ref inner) => inner.fmt(f),
+            _ => self.repr.fmt(f),
```
[the remark this answers is not in the document]
Comment (jan-auer) [reply symbolic-debuginfo/src/object.rs:101]: Good question. This always used to resolve correctly in the past, but it's probably worth disambiguating this.

File: symbolic-debuginfo/src/dwarf.rs [hunk 2]
```diff
@@ -51,29 +52,76 @@ fn offset(addr: u64, offset: i64) -> u64 {
...
-    #[error("unexpected inline function without parent")]
     UnexpectedInline,
 
     /// The debug_ranges of a function are invalid.
-    #[error("function with inverted address range")]
     InvertedFunctionRange,
 
     /// The DWARF file is corrupted. See the cause for more information.
-    #[error("corrupted dwarf debug data")]
-    CorruptedData(#[from] GimliError),
```
Comment (Swatinem) [inline symbolic-debuginfo/src/dwarf.rs:76]: `GimliError` was exposed because of the pub usage here. Maybe we can hide all the `gimli` re-exports again?

Comment (jan-auer) [reply symbolic-debuginfo/src/dwarf.rs:76]: Good catch, I'll create a follow-up.

Comment (jan-auer) [reply symbolic-debuginfo/src/dwarf.rs:76]: https://github.com/getsentry/symbolic/pull/300

**Pull request merged**